src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isPresent()
+                && user.get().getPassword().equals(HashUtil.hashPassword(password))) {
+            return user.get();
+        }
+        throw new AuthenticationException(
+                String.format("Email or password aren't valid "
+                        + "or user with email=%s wasn't registered", email));
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
```

Comment:
now when you delete password length validation which exception can you get?

Reply:
forgot